node/pkg/watchers/algorand/watcher.go

```diff
@@ -182,79 +180,123 @@ func (e *Watcher) Run(ctx context.Context) error {
 
 	logger.Info(fmt.Sprintf("first block %d", e.next_round))
 
-	for {
-		select {
-		case <-ctx.Done():
-			return nil
-		case r := <-e.obsvReqC:
-			if vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
-				panic("invalid chain ID")
-			}
+	// Create the timer for the getting the block height
+	timer := time.NewTicker(time.Second * 1)
+	defer timer.Stop()
 
-			logger.Info("Received obsv request",
-				zap.String("tx_hash", hex.EncodeToString(r.TxHash)),
-				zap.String("base32_tx_hash", base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)))
+	// Signal that basic initialization is complete
+	readiness.SetReady(e.readinessSync)
 
-			result, err := indexerClient.SearchForTransactions().TXID(base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)).Do(context.Background())
-			if err != nil {
-				logger.Error("SearchForTransactions", zap.Error(err))
-				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
-				break
-			}
-			for _, t := range result.Transactions {
-				r := t.ConfirmedRound
+	// Create an error channel
+	errC := make(chan error)
+	defer close(errC)
+
+	// Create the go routine to handle events from core contract
+	common.RunWithScissors(ctx, errC, "core_events_and_block_height", func(ctx context.Context) error {
+		logger.Info("Entering core_events_and_block_height...")
+
+		for {
+			select {
+			case err := <-errC:
+				logger.Error("core_events_and_block_height died", zap.Error(err))
+				return fmt.Errorf("core_events_and_block_height died: %w", err)
+			case <-ctx.Done():
+				logger.Error("core_events_and_block_height context done")
+				return ctx.Err()
 
-				block, err := algodClient.Block(r).Do(context.Background())
+			case <-timer.C:
+				status, err := algodClient.Status().Do(context.Background())
 				if err != nil {
-					logger.Error("SearchForTransactions", zap.Error(err))
+					logger.Error(fmt.Sprintf("algodClient.Status: %s", err.Error()))
```

Comment:
Why are you doing a sprintf here rather than just using `zap.Error(err)`? Also, if you are going to do the sprintf, I think you should use %w to log error.

Reply:
Fixed